### DISCUSSION TITLE
Bluetooth Usage | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1170] [Q&A]
# Bluetooth Usage | Seeed Studio Wiki

Bluetooth Usage on Seeed Studio XIAO ESP32C3

https://wiki.seeedstudio.com/XIAO_ESP32C3_Bluetooth_Usage/

**Comment by BFree42** [comment]
I can't get the Scan Bluetooth Devices sample code to work. I'm completely new to Bluetooth programming, so I apologize if I'm missing something simple.
I have a XIAO ESP32-C3 arduino. I've followed the instructions on this page () for setting it up in the Arudino IDE.

I copied and pasted the above sample code into a sketch, and when I tried to verify it, here's the error message I got:

C:\Users\User\Documents\Arduino\ESP32C3_ScanForDevices_VerIDE\ESP32C3_ScanForDevices_VerIDE.ino: In function 'void loop()':
C:\Users\User\Documents\Arduino\ESP32C3_ScanForDevices_VerIDE\ESP32C3_ScanForDevices_VerIDE.ino:29:48: error: conversion from 'BLEScanResults*' to non-scalar type 'BLEScanResults' requested
   29 |   BLEScanResults foundDevices = pBLEScan->start(scanTime, false);
      |                                 ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~

exit status 1

Compilation error: conversion from 'BLEScanResults*' to non-scalar type 'BLEScanResults' requested

How do I fix this? Thank you!!!

**Reply by KronbergStudios** [reply]
Change it to:

BLEScanResults * foundDevices = pBLEScan->start(scanTime, false);

**Comment by w0gen** [comment]
I am getting the same error!  What library should we be using?

**Reply by BFree42** [reply]
I gave up on the sample code that's on this page, but if you follow the directions on the "Getting Started with Seeed Studio XIAO ESP32C3" page to set up the Arduino IDE, then in the IDE click the "File" menu, and then hover over "Examples." There should be a whole section in that second menu dedicated to XIAO_ESP32C3. If you go to the "BLE" folder in that secondary menu, there are other example codes that do compile and work. Hope this helps!
Good luck!

**Reply by KronbergStudios** [reply]
There are two changes required to run samples:

For the first one:
BLEScanResults * foundDevices = pBLEScan->start(scanTime, false);

For the second one:
String value = pCharacteristic->getValue();

**Comment by vakkerman** [comment]
Hello! I need help with BTHome integration for Home Assistant. How can I send data from esp32c6 via BLE? Or Passive BLE - another integration.

**Reply by nfs0619** [reply]
Hello! Thanks so much for reaching out to us. For integrating BTHome with your ESP32-C6 via BLE, I’d recommend checking out our community forum at [https://forum.seeedstudio.com/](https://forum.seeedstudio.com/). If you don’t find a solution there, you can always create a new post to ask your question and get help from other users who might have faced similar issues.

If you need more specific assistance or have further questions, don’t hesitate to get in touch with our tech support team at [techsupport@seeed.io](mailto:techsupport@seeed.io). We’re happy to help however we can!

**Comment by OldMicroGuy** [comment]
Has anybody actually tested the Blutooth server code??
I uploaded it exactly as it (after many hours of trying other "known good code", and like everything, it doesn't work.
I have code that works just fine on a CYD, burn that on to this XIAO chip, and it doesn't work, nor anything from the serial port.
Same thing with this sample code, no comms on anything.
I know the chip is good, because if I load the super basic starter that comes with a new project in PlatformIO, and add print statements, it prints like crazy.   So it seems these chips don't like BLE.

**Reply by OldMicroGuy** [reply]
Ok, apparently, the code works over BLE, but the serial port konks out.  I can't print diagnostics out through the serial port.

**Reply by Lukilyy** [reply]
Hello, enabling BLE on the ESP32-C3 causes a conflict with the built-in USB CDC, disabling the default serial port. This is a chip limitation, not a hardware design issue. We recommend using UART1 with an external USB-to-Serial module for debugging. We will also conduct tests on our side to verify this behavior.
